docs/sources/k6/next/javascript-api/k6-browser/page/on.md

```diff
@@ -7,24 +7,19 @@ description: 'Browser module: page.on method'
 
-| Parameter | Type     | Default | Description                                                                         |
-| --------- | -------- | ------- | ----------------------------------------------------------------------------------- |
-| event     | string   | `''`    | Event to attach the handler to. Currently, only the `'console'` event is supported. |
-| handler   | function | `null`  | A function to be called every time the specified event is emitted.                  |
-
-{{< admonition type="caution" >}}
-
-When using the `page.on` method, the page has to be explicitly [closed](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-browser/page/close/) for the iteration to be able to finish.
-
-{{< /admonition >}}
+| Parameter | Type     | Default | Description                                                                                 |
+| --------- | -------- | ------- | ------------------------------------------------------------------------------------------- |
+| event     | string   | `''`    | Event to attach the handler to. Currently, `'console'` and `'metric'` events are supported. |
+| handler   | function | `null`  | A function to be called every time the specified event is emitted.                          |
 
 ### Events
 
-| Event     | Description                                                                                                                                                                                                                                                                       |
-| --------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Event     | Description                                                                                                                                                                                                                                                          |
+| --------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | `console` | Emitted every time the console API methods are called from within the page JavaScript context. The arguments passed into the handler are defined by the [`ConsoleMessage`](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-browser/consolemessage) class. |
+| `metric`  | Emitted every time a metric is measured and emitted for the page. The arguments passed into the handler are defined by the [`MetricMessage`](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-browser/metricmessage) class.                                |
```

Comment:
To avoid confusion, I would rename "class" with "object" as we say a `MetricMessage` object in the introduction.

Reply:
Yep, makes sense, updated in f45cda8df1d61df4e201f0a01f6ec8194916ed27